src/sagas/index.ts

```diff
@@ -140,23 +140,65 @@ function* workspaceSaga(): SagaIterator {
     }
   })
 
+  /**
+   * Ensures that the external JS libraries have been loaded, with a
+   * timeout of 3 seconds. An error will be thrown if the libraries are not
+   * loaded.
+   */
+  function* checkWebGLAvailable() {
+    /**
+     * A generator function that constantly checks if getReadyWebGLForCanvas
+     * is available in the global (window) scope. Returns (true) if such a function
+     * exists.
+     */
+    function* helper() {
+      while (true) {
+        if ((window as any).getReadyWebGLForCanvas !== undefined) {
+          break
+        }
+        /** Prevent checking instantaenously by waiting for a bit. */
+        yield call(delay, 250)
+      }
+      return true
+    }
+    /** Create a race condition between the js files being loaded and a 3 second timeout. */
+    const { loadedScripts, timeout } = yield race({
+      loadedScripts: call(helper),
+      timeout: call(delay, 4000)
```

Comment:
The comment above should say 4s? Anyways I think the comments from lines 143 to here are a little unnecessary. Instead maybe comment (and not necessarily here),

- where does getReadyWebGLForCanvas come from?
- why is there a need to check if it's available (runtime errors)?